Sources/AWSSDKSwiftCore/HMAC.swift

```diff
@@ -6,19 +6,18 @@
 //
 //
 
-import Foundation
-import CNIOOpenSSL
+import CNIOBoringSSL
 
 func hmac(string: String, key: [UInt8]) -> [UInt8] {
     var context = HMAC_CTX()
-    HMAC_Init_ex(&context, key, Int32(key.count), EVP_sha256(), nil)
+    CNIOBoringSSL_HMAC_Init_ex(&context, key, key.count, CNIOBoringSSL_EVP_sha256(), nil)
```

Comment:
You probably don't want to use BOringSSL from NIO because its API will definitely be broken in the future. cc @weissi

Reply:
Hmmm, using the BoringSSL implementation means we don't need to link in another SSL lib. What are the plans that are going to break the API in the future?

Reply:
Im not sure, but most of the NIO team Ive talked to when they have reviewed my PRs, basically say 100% we shouldn't be using it. Heres one explanation. https://forums.swift.org/t/discussion-nioapns-nio-based-apple-push-notification-service/23384/11?u=kylebrowning

Reply:
As @kylebrowning is pointing out, please definitely don't use NIO's internal modules. The only public API is what's exported by the [Package.swift](https://github.com/apple/swift-nio-ssl/blob/master/Package.swift#L30) as `.library` targets which for `swift-nio-ssl` is just `NIOSSL`.

That we are vending BoringSSL is literally just an implementation detail and the API does change a lot. We have (I think 3 times now) re-namespaced BoringSSL and BoringSSL itself _does not have_ a public API. So it's not us being unhelpful it's that BoringSSL doesn't have an API.

[BoringSSL](https://boringssl.googlesource.com/boringssl/) also documents that fact itself:

> Although BoringSSL is an open source project, it is not intended for general use, as OpenSSL is. We don't recommend that third parties depend upon it. Doing so is likely to be frustrating because there are no guarantees of API or ABI stability.
>
> Programs ship their own copies of BoringSSL when they use it and we update everything as needed when deciding to make API changes. This allows us to mostly avoid compromises in the name of compatibility. It works for us, but it may not work for you.

Reply:
ok, will look for an alternative

Reply:
@weissi are we ok to use swift-nio-ssl-support.git to load libressl, or should we just implement our own?

Reply:
@adam-fowler yes, that's totally acceptable. Rule of thumb: Everything officially exported by a `Package.swift` is fine and `swift-nio-ssl-support` basically exports the system's libssl to you.